activesupport/lib/active_support/callbacks.rb

```diff
@@ -826,6 +754,30 @@ def get_callbacks(name)
       def set_callbacks(name, callbacks)
         send "_#{name}_callbacks=", callbacks
       end
+
+      def default_false_terminator
+        Proc.new do |target, result_lambda|
+          terminate = true
+          catch(:abort) do
+            result = result_lambda.call if result_lambda.is_a?(Proc)
+            if CallbackChain.halt_and_display_warning_on_return_false && result == false
```

Comment:
Because in ActiveSupport <= 4.2, the value returned by callbacks was (already) ignored.

Therefore, the behavior in Rails 5 does not change for ActiveSupport itself.

It only changes for AR and AM that were using ActiveSupport::Callbacks, but specifying a custom terminator that said "if any callback returns `false`, halt the chain".

Therefore, if someone returns `false` in AR or AM callbacks in Rails 5, we have to *warn* them that they should change their code if they really want the chain to be halted, otherwise their code will not have any effect starting with Rails 5.1.